src/axom/sidre/core/Group.hpp

```diff
@@ -293,35 +294,36 @@ class Group
    */
   Group* setDefaultAllocator(umpire::Allocator alloc)
   {
-    m_default_allocator_id = alloc.getId();
+    setDefaultAllocator(alloc.getId());
     return this;
   }
+#endif
 
   /*!
    * \brief Set the default umpire::Allocator associated with this Group.
    */
   Group* setDefaultAllocator(int allocId)
   {
+    SLIC_ASSERT(allocId != axom::INVALID_ALLOCATOR_ID);
     m_default_allocator_id = allocId;
     return this;
   }
-#else
-  /*!
-   * \brief Return the ID of the default umpire::Allocator associated with this
-   * Group.
-   */
-  int getDefaultAllocatorID() const { return axom::getDefaultAllocatorID(); }
 
   /*!
-   * \brief Set the default umpire::Allocator associated with this Group.
+   * \brief Reallocate data to View-specific allocators.
+   *
+   * \param [i] viewToAllocatorId A function that returns the allocator
+   *   id to reallocate the View's data to.  Where it returns
+   *   the View's current allocator or axom::INVALID_ALLOCATOR_ID,
+   *   the View is unchanged.
+   *
+   * \return pointer to this Group object.
+   *
+   * This method is recursive.
```

Comment:
Does `recursive` mean that the operation is applied to the entire group hierarchy rooted at the group?
If so, maybe it's clearer to say something like that. If not, I would have to look at the implementation to see what it means...

Reply:
The method is no longer needed after the shift toward supporting tuples and has been removed.